packages/NodeTypeResolver/NodeTypeResolver.php

```diff
@@ -139,6 +140,13 @@ public function resolve(Node $node): Type
 
     public function getType(Node $node): Type
     {
+        if ($node instanceof NullableType) {
+            $type = $this->getType($node->type);
+            if (! $type instanceof MixedType) {
+                return new UnionType([$type, new NullType()]);
+            }
+        }
+
```

Comment:
:+1: 

Btw, what Type is produced without this change?

Reply:
it was mixed type without this check.